pkg/agent/flowexporter/connections/conntrack_linux.go

```diff
@@ -150,3 +154,19 @@ func netlinkFlowToAntreaConnection(conn *conntrack.Flow) *flowexporter.Connectio
 
 	return &newConn
 }
+
+func setupConntrackParameters() error {
+	// Ensure net.netfilter.nf_conntrack_acct value to be 1. This will enable flow exporter to export stats of connections.
+	// Do not fail, but continue after logging error as we can still dump flows with no stats.
+	acctErr := sysctl.EnsureSysctlNetValue("netfilter/nf_conntrack_acct", 1)
+	if acctErr != nil {
+		return acctErr
+	}
+	// Ensure net.netfilter.nf_conntrack_timestamp value to be 1. This will enable flow exporter to export timestamps of connections.
+	// Do not fail, but continue after logging error as we can still dump flows with no timestamps.
+	timestampErr := sysctl.EnsureSysctlNetValue("netfilter/nf_conntrack_timestamp", 1)
+	if timestampErr != nil {
+		return timestampErr
+	}
+	return nil
+}
```

Comment:
@srikartati the comments no longer seem to match the implementation. What are your suggestions?

I feel like this got a bit messy because of testing. What we could do is:

 * change the implementation of this function

```go
func setupConntrackParameters() error {
    parametersWithErrors := []string{}
    if sysctl.EnsureSysctlNetValue("netfilter/nf_conntrack_acct", 1) != nil {
        parametersWithErrors = append(parametersWithErrors, "net.netfilter.nf_conntrack_acct")
    }
    if sysctl.EnsureSysctlNetValue("netfilter/nf_conntrack_timestamp", 1) != nil {
        parametersWithErrors = append(parametersWithErrors, "net.netfilter.nf_conntrack_timestamp")
    }
    if len(parametersWithErrors) > 0 {
        return fmt.Errorf("the following kernel parameters could not be verified / set: %v", parametersWithErrors)
    }
    return nil
}
```

 * change the implementation of `NewConnTrackSystem`

```go
func NewConnTrackSystem(nodeConfig *config.NodeConfig, serviceCIDR *net.IPNet) *connTrackSystem {
    if err := setupConntrackParameters(); err != nil {
        // Do not fail, but continue after logging an error as we can still dump flows with missing information.
        klog.Errorf("Error when setting up conntrack parameters, some information may be missing from exported flows: %v", err)
    }
    return &connTrackSystem{...}
}
```

 * in the unit tests: call `setupConntrackParameters()` separately to determine whether these parameters can be set and therefore verified. This is a temporary solution since we will update this code in the future.

Reply:
Yes, I was thinking the same about the old comments.
Testing setupConntrackParameters separately is better in unit tests, and we could create the new object using &connTrackSystem{...} like before in unit tests. 

Pointer 1 is a good idea. Thanks for suggesting.

Reply:
Thanks for your advices. I finished changing and added a unit test for setupConntrackParameters() function.